Admin documentation (#633)
SwissDataScienceCenter/renku: docs/admin/prerequisites/loadbalancer.rst

.. _loadbalancer:

Configuring a Load Balancer
===========================

> To set up a load balancer for k8s in your cluster you can configure a node with an HAProxy instance or use LBASS from your cloud provider.

Reviewer: ```suggestion
To set up a load balancer for k8s in your cluster you can configure a node with an HAProxy instance or use LoadBalancer-as-a-Service (LBaaS) from your cloud provider.
```

> A. Using an HAProxy loadbalancer

Reviewer: ```suggestion
Using an HAProxy loadbalancer
```

---------------------------------

> Steps to setup and configure the HA proxy server.

Reviewer: I think this section is too switch-specific for the generic docs

         server <node3-hostname> <node3-ip>:32022 check port 32022 fall 3 rise 2

  #### port 443, https traffic
  listen https 0.0.0.0:443
         balance roundrobin
         mode tcp
         option tcp-check
         server <node1-hostname> <node1-ip>:32443 check port 32443 fall 3 rise 2
         server <node2-hostname> <node2-ip>:32443 check port 32443 fall 3 rise 2
         server <node3-hostname> <node3-ip>:32443 check port 32443 fall 3 rise 2

  ## Port 80 is open to all, to allow ACME challenge http01 to work
  #### port 80, http traffic
  listen http 0.0.0.0:80
         balance roundrobin
         mode http
         option httpchk HEAD /healthz  HTTP/1.0
         server <node1-hostname> <node1-ip>:32080 check port 32080 fall 3 rise 2
         server <node2-hostname> <node2-ip>:32080 check port 32080 fall 3 rise 2
         server <node3-hostname> <node3-ip>:32080 check port 32080 fall 3 rise 2

  #### port 6443, k8s API https traffic
  listen https 0.0.0.0:6443
         balance roundrobin
         mode tcp
         option tcp-check
         server <master-hostname> <master-ip>:6443 check port 6443 fall 3 rise 2

> B. Using an Openstack LBaaS loadbalancer

Reviewer: ```suggestion
Using an Openstack LBaaS loadbalancer
```
